common4j/src/test/com/microsoft/identity/common/java/net/UrlConnectionHttpClientTest.java

```diff
@@ -1092,4 +1106,129 @@ private void mockRequestBody(final HttpURLConnection mockedConnection) throws IO
     private URL getRequestUrl() throws MalformedURLException {
         return new URL("https://login.microsoftonline.com/common");
     }
-}
+
+    /**
+     * Note: The following tests are made against https://badssl.com
+     *       and they could change the URL/configuration at any time.
+     *       If these test fails, check the website first.
+     */
+
+    @Test
+    public void testNoSSL() throws IOException {
+        final HttpResponse response = sNoRetryClient.method(
+                HttpClient.HttpMethod.GET,
+                new URL("http://http.badssl.com/"),
+                new LinkedHashMap<String, String>(),
+                null,
+                null
+        );
+
+        Assert.assertEquals(200,response.getStatusCode());
+        Assert.assertEquals("", SSLSocketFactoryWrapper.getLastHandshakeTLSversion());
+    }
+
+    @Test
+//    @Ignore("Ignored because our pipeline doesn't support TLS1.0 and TLS1.1. This still can be run locally.")
+    public void testTLS1() throws IOException {
+        final HttpResponse response = sNoRetryClient.method(
+                HttpClient.HttpMethod.GET,
+                new URL("https://tls-v1-0.badssl.com:1010/"),
+                new LinkedHashMap<String, String>(),
+                null,
+                null
```

Comment:
Can we run these as methods that take an ssl context?

Reply:
@AdamBJohnsonx  I can add tests for that.